src/bp/core/routers/bots/index.ts

```diff
@@ -281,15 +281,52 @@ export class BotsRouter extends CustomRouter {
     )
 
     this.router.post(
-      '/flows',
+      '/flow',
       this.checkTokenHeader,
       this.needPermissions('write', 'bot.flows'),
       this.asyncMiddleware(async (req, res) => {
         const botId = req.params.botId
-        const flowViews = <FlowView[]>req.body.dirtyFlows
+        const flow = <FlowView>req.body.flow
 
-        await this.flowService.saveAll(botId, flowViews, req.body.cleanFlows)
-        res.sendStatus(201)
+        await this.flowService.upsertFlow(botId, flow)
+        res.sendStatus(200)
+      })
+    )
+
+    this.router.put(
+      '/flow/:flowName',
```

Comment:
Flows can contain slashes, and it would result in a 404 not found
![image](https://user-images.githubusercontent.com/42552874/61000768-210e8900-a32c-11e9-9124-ba2c90a2c153.png)

Reply:
Good point, I get around by replacing forward slashes by %2F.